Show a bar with the events to which the user contributes

amieiro opened the pull request:
This PR adds a bar with the events to which the user contributes.

If the user contributes to up to 3 events, it displays all of them. If she contributes to more, it shows 2 and a text with the number of the other events to which he/she contributes.

### User who does not attend any event

![Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-21-07](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/03173a35-292d-4981-8e5d-4b0335fb57e8)

### User attending 1 event

![Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-17-24](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/de52e638-3503-4c4e-9ff3-cd0272ae1284)

### User attending 2 events

![Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-16-46](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/f434862e-5ee5-45b0-b515-0246ef46c7b3)

### User attending 3 events

![Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-19-05](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/e54f22ed-d60f-4b7c-86ab-1c47c1b1db73)

### User attending more than 3 events

![Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-20-02](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/b84da5be-9b43-4b28-845a-fdc6a6100666)
 
This PR needs [this PR](https://github.com/GlotPress/GlotPress/pull/1792) to be approved.

Fixes https://github.com/WordPress/wporg-gp-translation-events/issues/98.

psrpinto commented on wporg-gp-translation-events.php:
```diff
@@ -372,3 +373,80 @@ function () {
+ * @return void
+ */
+function add_active_events_current_user(): void {
+	$user_attending_events      = get_user_meta( get_current_user_id(), Route::USER_META_KEY_ATTENDING, true ) ?: array();
```
I think maybe we should consider refactoring [the code that retrieves active events the user is attending](https://github.com/WordPress/wporg-gp-translation-events/blob/trunk/includes/translation-listener.php#L69-L70) to something reusable, as I noticed we need that in several places. 

If we would do that, we would automatically get caching for the query that retrieves active events, as that can be an expensive query.

akirk replied:
I agree that we should do this! Tracking this in #117

amieiro replied:
Ok. I think it is better to address this problem in another PR, as we have an issue to it.

psrpinto reviewed (commented):
Looks good. I'll test this once the GlotPress PR gets merged.

akirk commented on wporg-gp-translation-events.php:
```diff
@@ -372,3 +373,80 @@ function () {
+	} else {
+		while ( $user_attending_events_query->have_posts() ) {
+			$user_attending_events_query->the_post();
+			$content .= '<span class="active-events-before-translation-table">' . get_the_title() . '</span>';
```
Can we make it a link? Maybe refactor this to use a function.

amieiro replied:
Done [here](https://github.com/WordPress/wporg-gp-translation-events/pull/114/commits/091bc7e8a7cdacf6ed278a4a8e727dc479f64f8f).

akirk commented on wporg-gp-translation-events.php:
```diff
@@ -372,3 +373,80 @@ function () {
+		$counter = 0;
+		while ( $user_attending_events_query->have_posts() && $counter < 2 ) {
+			$user_attending_events_query->the_post();
+			$content .= '<span class="active-events-before-translation-table">' . get_the_title() . '</span>';
```
Can we make it a link?

amieiro replied:
Done [here](https://github.com/WordPress/wporg-gp-translation-events/pull/114/commits/091bc7e8a7cdacf6ed278a4a8e727dc479f64f8f).

akirk reviewed (commented):
Nice, my only remark is that it looks a bit too much like a warning for my taste but I am curious for other opinions.

amieiro commented:
> Nice, my only remark is that it looks a bit too much like a warning for my taste but I am curious for other opinions.

What about changing from `--gp-color-attention-emphasis` to `--gp-color-status-waiting-subtle`:

### Current color

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/71437f4f-17f4-4383-974e-4f4f18c714e6)

### New color

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/61fa26f1-b6be-499f-9a2f-90583c51c56a)

amieiro commented:
I have added the links to each event in the bubbles and a link to the main page in the "and 2 more events", because the user can see the "Event she is attending" in this page, and we don't have a page with only this information.

I have changed the background color to a lighter one. I prefer to use a GlotPress color, to avoid color inconsistencies if they are changed in the future.

@psrpinto the hook in GlotPress has been merged, so you can pull the `develop` branch from GlotPress and test it.

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/f17e179f-f4b1-469d-af5c-36b4710f1959)

akirk reviewed (approved):
Works well after pulling the latest develop branch of GlotPress. Maybe we can ask a designer later on to take another look at the styling, also in general of the plugin.

psrpinto reviewed (approved):
Tested and it works 👌
